src/coreclr/jit/codegenxarch.cpp

```diff
@@ -2457,7 +2457,7 @@ void CodeGen::genLclHeap(GenTree* tree)
 
         // For small allocations we will generate up to six push 0 inline
         size_t cntRegSizedWords = amount / REGSIZE_BYTES;
-        if (cntRegSizedWords <= 6)
+        if (compiler->info.compInitMem && (cntRegSizedWords <= 6))
```

Comment:
I'm not aware of this code, but which value is `compiler->info.compInitMem`?
Is it 20 for `stackalloc byte[20]` or already rounded to some greater value?

I'm asking because of the asserts in L2455-L2456 (just above). They'd fail for 20.

Reply:
those assert pass because 20 (or 40 in bytes) is aligned to 24 (48) here first: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/codegenxarch.cpp#L2354